### PR TITLE
deps: move from github.com/pkg/errors to 'errors' and 'fmt'

### PR DESCRIPTION
**Pull request by mmorel-35**
**Description:** 
The package https://github.com/pkg/errors is not maintained anymore.
This PR replaces https://github.com/pkg/errors with official errors and fmt golang packages

Signed-off-by: Matthieu MOREL [mmorel-35@users.noreply.github.com](mailto:mmorel-35@users.noreply.github.com)

### CODE REVIEW

File: receiver/k8sclusterreceiver/receiver.go [hunk 1]
```diff
@@ -115,11 +115,12 @@ func (kr *kubernetesReceiver) dispatchMetrics(ctx context.Context) {
...
 func newReceiver(
 	set component.ReceiverCreateSettings, config *Config, consumer consumer.Metrics,
-	client kubernetes.Interface, osQuotaClient quotaclientset.Interface) (component.MetricsReceiver, error) {
+	client kubernetes.Interface, osQuotaClient quotaclientset.Interface,
+) (component.MetricsReceiver, error) {
 	resourceWatcher, err := newResourceWatcher(set.Logger, client, osQuotaClient, config.NodeConditionTypesToReport,
 		config.AllocatableTypesToReport, defaultInitialSyncTimeout)
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to setup the receiver")
+		return nil, fmt.Errorf("Failed to setup the receiver: %w", err)
```
Comment (bogdandrutu) [inline receiver/k8sclusterreceiver/receiver.go:123]: lower case first letter.

File: receiver/k8sclusterreceiver/watcher.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func (rw *resourceWatcher) isKindSupported(gvk schema.GroupVersionKind) (bool, e
 			rw.logger.Debug("Group version is not supported", zap.String("group", gvk.GroupVersion().String()))
 			return false, nil
 		}
-		return false, errors.Wrap(err, "Failed to fetch group version details")
+		return false, fmt.Errorf("Failed to fetch group version details: %w", err)
```
Comment (bogdandrutu) [inline receiver/k8sclusterreceiver/watcher.go:144]: lower case first letter.

**Review by bogdandrutu — APPROVED** [review]

File: receiver/k8sclusterreceiver/receiver.go (hunk 1 above)
Comment (mmorel-35) [reply receiver/k8sclusterreceiver/receiver.go]: Ok ! Done !

File: receiver/k8sclusterreceiver/watcher.go (hunk 2 above)
Comment (mmorel-35) [reply receiver/k8sclusterreceiver/watcher.go]: Ok ! Done !

**Comment by mx-psi** [comment]
@mmorel-35 can you fix the merge conflict?

**Comment by mmorel-35** [comment]
Hi @mx-psi ,
The conflict has been handled.

**Review by dmitryax — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+- `k8sclusterreceiver`: move from github.com/pkg/errors to 'errors' and 'fmt' (#11114)
+- `mongodbatlasreceiver`: move from github.com/pkg/errors to 'errors' and 'fmt' (#11114)
+
```
Comment (dmitryax) [inline CHANGELOG.md:7]: It's not a user facing change, so no need to mention in the changelog. Can you remove this block? I'll add the "Skip changelog" label to pass CI.

Comment (mmorel-35) [reply CHANGELOG.md]: Hi @dmitryax ,
I fixed it.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**